### PR TITLE
fix(log_to_metric transform): set not working in log-to-metric transform when all_metrics=true

### PR DESCRIPTION
**Pull request by pabloem**
I am pretty sure this is a bug. I was not able to make this work. I submit a test case that fails on current released versions. How could I set up this test case to run continuously?

### CODE REVIEW

**Review by tobz — CHANGES_REQUESTED** [review]
Thanks for your contribution here. 👍🏻 

As far as adding a test for it, the `testing` folder is actually meant simply to hold Vector configurations and test case documentation used when running tests _manually_.

You'll want to to create a new unit test in `src/transforms/log_to_metric.rs` in this case.

**Comment by jszwedko** [comment]
There are also high-level behavior tests that run here: https://github.com/vectordotdev/vector/tree/master/tests/behavior/transforms. I agree that a unit test seems the most appropriate for this change though.

**Comment by pabloem** [comment]
okay I added a test case. @tobz - let me know if that looks good.

Once we merge, can you also let me know in which release of vector the PR would land?

**Comment by jszwedko** [comment]
> Once we merge, can you also let me know in which release of vector the PR would land?

It would be in v0.38.0 currently scheduled for May 7th. It'll be in the nightly builds earlier though if you want to try it out.

**Review by jszwedko — COMMENTED** [review]
Thanks for the test! Do you mind adding a changelog fragment describing the change too? See https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md

**Comment by pabloem** [comment]
> Thanks for the test! Do you mind adding a changelog fragment describing the change too? See https://github.com/vectordotdev/vector/blob/master/changelog.d/README.md

added! thanks!

File: changelog.d/20228_fix-log-to-metric-set-supported.fix.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Fixed an issue where the log_to_metric transform with all_metrics=true config failed to convert properly-formatted 'set'-type events into metrics.
+
+author: Pablo E.
```
Comment (jszwedko) [inline changelog.d/20228_fix-log-to-metric-set-supported.fix.md:3]:
```suggestion
author: pabloem
```

Comment (jszwedko) [reply changelog.d/20228_fix-log-to-metric-set-supported.fix.md:3]: This is actually expected to be a GH username.

**Review by jszwedko — COMMENTED** [review]
Thanks @pabloem! Just one comment about the changelog, but otherwise this looks good.

**Comment by pabloem** [comment]
Done thanks @jszwedko

**Review by jszwedko — APPROVED** [review]

**Comment by pabloem** [comment]
looks like @tobz needs to change his review status? : ) - I don't know how to tell it to ignore the spellcheck on my username.

File: changelog.d/20228_fix-log-to-metric-set-supported.fix.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Fixed an issue where the log_to_metric transform with all_metrics=true config failed to convert properly-formatted 'set'-type events into metrics.
+
+author: pabloem
```
Comment (jszwedko) [inline changelog.d/20228_fix-log-to-metric-set-supported.fix.md:3]:
```suggestion
authors: pabloem
```

Just realized this should be `authors:` which should also tell the spell checker to ignore the line.

**Comment by jszwedko** [comment]
I'll let @tobz give this another 👍 in case there was something else he was expecting.

**Review by tobz — APPROVED** [review]

**Pull request merged**